src/hermit.rs

```diff
@@ -43,11 +43,38 @@ impl<T: Config> Hermit<T> {
         let path = self.config.shell_root_path().join(name);
         file_ops.create_git_repo(path);
     }
+
+    pub fn inhabit_shell(&mut self, file_ops: &mut FileOperations, name: &str) -> Result<Vec<String>, Error> {
+        let mut shell_names = Vec::new();
+        let mut res = Ok(());
+        if name == "" {
+            shell_names = self.get_shell_list();
+        } else {
+            res = self.set_current_shell(name);
+        }
+        match res {
+            Ok(v) => Ok(shell_names),
+            Err(e) => Err(Error::ShellDoesNotExist),
+        }
+    }
+
+    pub fn get_shell_list(&mut self) -> Vec<String> {
+        let mut shell_names = Vec::new();
+        let paths = fs::read_dir(self.config.shell_root_path().to_owned()).unwrap();
+        for path in paths {       
+            let shell_path = path.unwrap().path();
+            let shell_name = shell_path.file_name().unwrap().to_str().unwrap().to_owned();
+            shell_names.push(shell_name);
+            shell_names.sort();
```

Comment:
you probably only want to call sort after you've collected all the shell names...